### PR TITLE
SONARJAVA-1075 Remove usage of ASTNode in RightCurlyBraceStartLineCheck

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java [hunk 1]
```diff
@@ -41,43 +61,153 @@
...
-    return node.getParent().is(Kind.NEW_ARRAY);
+  private SyntaxToken getLastToken(StatementTree lastStatement) {
+    StatementLastTokenFinder visitor = new StatementLastTokenFinder();
+    lastStatement.accept(visitor);
+    return visitor.lastToken;
   }
 
-  private static Token getPreviousToken(AstNode node) {
-    AstNode result = node.getPreviousAstNode();
+  private class StatementLastTokenFinder extends BaseTreeVisitor {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java]: :new: MAJORShould org.sonar.java.checks.RightCurlyBraceStartLineCheck$StatementLastTokenFinder be a _static_ inner class?[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASIC_INNER_SHOULD_BE_STATIC)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 major

Details:
* [Should org.sonar.java.checks.RightCurlyBraceStartLineCheck$StatementLastTokenFinder be a _static_ inner class?](https://github.com/SonarSource/sonar-java/blob/ae8565d9679ef319d67a60208aa20d49677028b9/java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java#L97)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ASIC_INNER_SHOULD_BE_STATIC)

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 2]
```diff
@@ -99,18 +106,23 @@ public MethodTreeImpl(
...
   }
 
-  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName) {
+  public MethodTreeImpl complete(TypeTree returnType, IdentifierTree simpleName, SyntaxToken semicolonToken) {
     Preconditions.checkState(this.simpleName == null);
     this.returnType = returnType;
     this.simpleName = simpleName;
+    this.semicolonToken = semicolonToken;
 
     prependChildren((AstNode) returnType, (AstNode) simpleName);
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: MAJORorg/sonar/java/model/declaration/MethodTreeImpl must not use com/sonar/sslr/api/AstNode[rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)

File: java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java [hunk 3]
```diff
@@ -41,43 +64,186 @@
...
-  }
+    @Override
+    public void visitCatch(CatchTree tree) {
+      scan(tree.block());
+    }
+
+    @Override
+    public void visitMethod(MethodTree tree) {
+      SyntaxToken semicolonToken = tree.semicolonToken();
+      if (semicolonToken != null) {
```
Comment (SonarTech) [inline java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java]: NewCRITICALRedundant nullcheck of semicolonToken, which is known to be non-null in org.sonar.java.checks.RightCurlyBraceStartLineCheck$SyntaxTokenFinder.visitMethod(MethodTree)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 new issue:
* 1 critical

Details:
* [Redundant nullcheck of semicolonToken, which is known to be non-null in org.sonar.java.checks.RightCurlyBraceStartLineCheck$SyntaxTokenFinder.visitMethod(MethodTree)](https://github.com/SonarSource/sonar-java/blob/4f82899a91acf78757985c0ec5161b39e38ec665/java-checks/src/main/java/org/sonar/java/checks/RightCurlyBraceStartLineCheck.java#L222)[rule](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3ARCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE)

**Comment by m-g-sonar** [comment]
Push on PR did not reorder correctly commits. I close this PR and open a new one

**Pull request closed without merging**